desktop/libs/notebook/src/notebook/static/notebook/js/notebook.ko.js

```diff
@@ -1785,16 +1789,24 @@ var EditorViewModel = (function() {
       var sessionJson = ko.mapping.toJSON(session);
 
       self.closeSession (session, true, function() {
+        //Removing the session is apparently the key to getting KO to have the right mapping
+        self.sessions.remove(session);
```

Comment:
How about keeping this line and instead of the other spinner logic tweaking cf. below?